parser/prism/Translator.cc

```diff
@@ -2001,12 +2013,13 @@ template <typename PrismNode> unique_ptr<parser::Mlhs> Translator::translateMult
 }
 
 // Context management methods
-Translator Translator::enterMethodDef() {
+Translator Translator::enterMethodDef() const {
+    auto resetDesugarUniqueCounter = true;
     auto isInMethodDef = true;
-    return Translator(parser, ctx, parseErrors, directlyDesugar, isInMethodDef, uniqueCounter);
```

Comment:
We're going to be introducing more context methods like `enterBlock`, `enterModule`, etc. Passing all the fields is annoying, so we pass `this` and have the constructor copy all the fields for us.

Reply:
Elsewhere we've done this by having instance methods that return a copy with the needed changes.

https://github.com/sorbet/sorbet/blob/91288c74bd5dae0850dc19992c0237a737a311ff/cfg/builder/builder.h#L60-L64